tests/test_edge_cases.py

```diff
@@ -527,7 +527,7 @@ class Foo(BaseModel):
 
     class Bar(Foo):
         x: float = 12.3
-        a = 123
+        a: int = 123
```

Comment:
This is a perhaps-unfortunate side effect -- this change would mean that you couldn't replace a float-annotated value with a default value that parses to `int` unless you reannotate.

Given that the check happens at class creation time though, this seems fine to me, and probably preferable over not erroring until instantiation in other cases.

Reply:
```suggestion
        a = 123.0
```

I think this is fine, but the above would be the canonical way of fixing it in this case I think.